deployment/zero-downtime-restart.md

```diff
@@ -0,0 +1,48 @@
+# Zero-downtime restart
+
+This feature allows a complete restart of Fluentd without bringing down some input plugins.
+
+Supported standard input plugins are as follows.
+
+| supported input plugin | version |
+| :---                   | :---    |
+| in_udp                 | v1.18.0 |
+| in_tcp                 | v1.18.0 |
+| in_syslog              | v1.18.0 |
+
+If these input plugins are down, client applications may fail to send data.
+If that client does not have a resend feature, the data will be lost.
+
+You can use this feature to completely restart Fluentd without losing data for these plugins.
+
+## How to use this feature
+
+You can use this feature in the following ways.
+
+* [Signals - SIGUSR2](signals.md#sigusr2)
+* [RPC](rpc.md)
+
+## Mechanism of zero-downtime restart
+
+![zero-downtime restart mechanism](../.gitbook/assets/fluentd-zero-downtime-restart-mechanism.png)
+
+1. The old supervisor receives `SIGUSR2`.
+2. Spawn a new supervisor.
+3. Take over shared sockets.
+4. Launch new workers, and stop old processes in parallel.
+   * Launch new workers with [Source Only Mode](source-only-mode.md).
+     * In addition to source-only mode, further limit the starting pluings to only those that support this feature.
+     * Data received by the new workers are stored in the temporary buffer of source-only mode.
+   * Send `SIGTERM` to the old supervisor after `10s` delay.
+5. The old supervisor stops and sends `SIGWINCH` to the new one.
+6. The new workers starts to run fully.
+   * The temporary buffer of source-only mode starts to load.
+
+You can configure the temporary buffer.
```

Comment:
The context "when need to be configured" seems missing. 
A bit more explanation is needed here.

Reply:
Is it something like this?

```diff
- You can configure the temporary buffer.
+ You can configure the temporary buffer in advance.
```

Reply:
I see! Certainly, it was hard to understand the context of the buffer.

I noticed it has been already explained in https://docs.fluentd.org/deployment/source-only-mode#temporary-file-buffer.
So, I fixed this as 17d31346d17c4164582e2c7f559002269d0a41a7.
How about this?